### PR TITLE
1474: Correctly display the location status on the location button

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Correctly display the location status on the location button and add new follow user icon.

### Proposed changes

- Extract LocationIcon component to correctly display status
- Add state to home_page to control the user follow mode
- Remove redundant locationFeatureEnabled setting (controlled by permission now)
- Remove onFeatureEnable and onFeatureDisable settings

### Side effects

None.

### Testing

- Test the location button in various settings (permissions granted, forbidden, ...)
- Move the map and recenter again
- Test the sorting button in various settings (permissions granted, forbidden, ...)

### Resolved issues

Fixes: #1474.

### CODE REVIEW

**Comment by steffenkleinle** [comment]
> Nicely done 👍 I've tested on android and ios and found one issue on android.
> 
> If i enable location permission, close the app and restart the app the map centers around my location but the location icon does not reflect that. Here two screen from android and ios
> If i click again on the location icon everything is fine but not initially when i start the app Tested on emulator pixel7 api level 33

Thanks, could you perhaps test again? Should be fixed now.

File: frontend/lib/map/location_button.dart [hunk 1]
```diff
@@ -1,31 +1,65 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    if (locationStatus == null) {
+      return const SmallButtonSpinner();
+    }
+
+    if (locationStatus == LocationStatus.always || locationStatus == LocationStatus.whileInUse) {
```
Comment (ztefanie) [inline frontend/lib/map/location_button.dart:21]: Naming the booleans here, would make this clearer

File: frontend/lib/map/location_button.dart [hunk 2]
```diff
@@ -1,31 +1,65 @@
...
+      if (followUserLocation) {
+        return Icon(Icons.my_location, color: theme.colorScheme.secondary);
+      }
+      return Icon(Icons.location_searching, color: theme.colorScheme.secondary);
+    }
```
Comment (ztefanie) [inline frontend/lib/map/location_button.dart:26]:
```suggestion
   final bool hasLocationPermission = locationStatus == LocationStatus.always || locationStatus == LocationStatus.whileInUse;

if (hasLocationPermission) {
  return Icon(
    followUserLocation ? Icons.my_location : Icons.location_searching,
    color: theme.colorScheme.secondary,
  );
}
```

File: frontend/lib/map/map/map_with_futures.dart [hunk 3]
```diff
@@ -30,13 +32,13 @@ class MapWithFutures extends StatelessWidget {
...
 
         return MapContainer(
-          onFeatureClick: onFeatureClick,
-          onNoFeatureClick: onNoFeatureClick,
-          locationAvailable: position?.isAvailable() ?? false,
-          userLocation: position?.toLatLng(),
-          onFeatureClickLayerFilter: onFeatureClickLayerFilter,
-          onMapCreated: onMapCreated,
-        );
+            onFeatureClick: onFeatureClick,
```
Comment (ztefanie) [inline frontend/lib/map/map/map_with_futures.dart:35]: Please undo the intendation change or is there a reason for it?

**Review by ztefanie — APPROVED** [review]
Looks good :+1: , just two small comments.

not tested

**Review by f1sh1918 — APPROVED** [review]
Works fine, tested on ios and android

File: frontend/lib/map/location_button.dart (hunk 2 above)
Comment (steffenkleinle) [reply frontend/lib/map/location_button.dart:26]: Nice :)

File: frontend/lib/map/map/map_with_futures.dart (hunk 3 above)
Comment (steffenkleinle) [reply frontend/lib/map/map/map_with_futures.dart:35]: Sadly had to redo, otherwise checks are failing :/

**Pull request merged**